### PR TITLE
do a null check before parsing the last timestamp of a event object

### PR DESCRIPTION
**Pull request by sankarpn**
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/9805
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/9806
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/9807
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/9842

### CODE REVIEW

**Review by anpanigr — APPROVED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java [hunk 1]
```diff
@@ -973,12 +975,10 @@ private static boolean isLocalPortFree(int port) {
     } catch (IOException ignored) {
       return true;
     } finally {
-      if (socket != null) {
-        try {
-          socket.close();
-        } catch (IOException ex) {
-          logger.severe("can not close Socket {0}", ex.getMessage());
-        }
+      try {
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java:978]: Got a feedback from the team to try to close the socket without a null check.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java [hunk 2]
```diff
@@ -940,6 +941,7 @@ public static synchronized int getNextFreePort() {
     while (port <= END_PORT) {
       freePort = port++;
       if (isLocalPortFree(freePort)) {
+        assertDoesNotThrow(() -> TimeUnit.SECONDS.sleep(3));
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java:944]: Got a feedback from team to wait for few seconds after the socket.close() call to give enough time to release the port.

**Review by vanajamukkara — APPROVED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java (hunk 1 above)
Comment (rjeberhard) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java:978]: This isn't the right pattern. You want: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Put the socket constructor in the try statement.

**Review by rjeberhard — CHANGES_REQUESTED** [review]
Please see the comment about try-with-resources.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java (hunk 1 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java]: Good suggestion. fixed it.

**Review by robertpatrick — APPROVED** [review]

**Comment by robertpatrick** [comment]
Please back port all relevant test changes to `release/3.4`

**Pull request merged**